File: backend/internal/domain/enterprise_polish.go
package domain

import "time"

type GovernanceRiskStatus string

const (
	GovernanceRiskStatusOpen       GovernanceRiskStatus = "open"
	GovernanceRiskStatusReviewing  GovernanceRiskStatus = "reviewing"
	GovernanceRiskStatusMitigating GovernanceRiskStatus = "mitigating"
	GovernanceRiskStatusClosed     GovernanceRiskStatus = "closed"
)

type GovernanceReportStatus string

const (
	GovernanceReportStatusDraft      GovernanceReportStatus = "draft"
	GovernanceReportStatusGenerating GovernanceReportStatus = "generating"
	GovernanceReportStatusReady      GovernanceReportStatus = "ready"
	GovernanceReportStatusExported   GovernanceReportStatus = "exported"
	GovernanceReportStatusArchived   GovernanceReportStatus = "archived"
)

type DeliveryArtifactStatus string

const (
	DeliveryArtifactStatusDraft      DeliveryArtifactStatus = "draft"
	DeliveryArtifactStatusActive     DeliveryArtifactStatus = "active"
	DeliveryArtifactStatusSuperseded DeliveryArtifactStatus = "superseded"
	DeliveryArtifactStatusArchived   DeliveryArtifactStatus = "archived"
)

type DeliveryReadinessConclusion string

const (
	DeliveryReadinessNotReady      DeliveryReadinessConclusion = "not-ready"
	DeliveryReadinessConditionally DeliveryReadinessConclusion = "conditionally-ready"
	DeliveryReadinessReady         DeliveryReadinessConclusion = "ready"
)

type GovernanceActionStatus string

const (
	GovernanceActionStatusOpen       GovernanceActionStatus = "open"
	GovernanceActionStatusInProgress GovernanceActionStatus = "in-progress"
	GovernanceActionStatusBlocked    GovernanceActionStatus = "blocked"
	GovernanceActionStatusResolved   GovernanceActionStatus = "resolved"
	GovernanceActionStatusClosed     GovernanceActionStatus = "closed"
)

type PermissionChangeTrail struct {
	ID                   uint64    `gorm:"primaryKey" json:"id"`
	WorkspaceID          uint64    `gorm:"index" json:"workspaceId"`
	ProjectID            *uint64   `gorm:"index" json:"projectId,omitempty"`
	SubjectType          string    `gorm:"size:64;not null" json:"subjectType"`
	SubjectRef           string    `gorm:"size:128;not null" json:"subjectRef"`
	SourceIdentity       string    `gorm:"size:128" json:"sourceIdentity"`
	ChangeType           string    `gorm:"size:64;not null" json:"changeType"`
	BeforeState          string    `gorm:"type:text" json:"beforeState"`
	AfterState           string    `gorm:"type:text" json:"afterState"`
	AuthorizationBasis   string    `gorm:"type:text" json:"authorizationBasis"`
	ApprovalReference    string    `gorm:"type:text" json:"approvalReference"`
	ScopeType            string    `gorm:"size:64;not null" json:"scopeType"`
	ScopeRef             string    `gorm:"size:128;not null" json:"scopeRef"`
	EvidenceCompleteness string    `gorm:"size:32;not null" json:"evidenceCompleteness"`
	ChangedAt            time.Time `gorm:"index" json:"changedAt"`
	ChangedBy            uint64    `gorm:"index;not null" json:"changedBy"`
	CreatedAt            time.Time `json:"createdAt"`
	UpdatedAt            time.Time `json:"updatedAt"`
}

func (PermissionChangeTrail) TableName() string { return "enterprise_permission_change_trails" }

type KeyOperationTrace struct {
	ID             uint64    `gorm:"primaryKey" json:"id"`
	WorkspaceID    uint64    `gorm:"index" json:"workspaceId"`
	ProjectID      *uint64   `gorm:"index" json:"projectId,omitempty"`
	ActorType      string    `gorm:"size:64;not null" json:"actorType"`
	ActorRef       string    `gorm:"size:128;not null" json:"actorRef"`
	OperationType  string    `gorm:"size:64;not null" json:"operationType"`
	TargetType     string    `gorm:"size:64;not null" json:"targetType"`
	TargetRef      string    `gorm:"size:128;not null" json:"targetRef"`
	ContextSummary string    `gorm:"type:text" json:"contextSummary"`
	RiskLevel      string    `gorm:"size:32;not null" json:"riskLevel"`
	Outcome        string    `gorm:"size:32;not null" json:"outcome"`
	OccurredAt     time.Time `gorm:"index" json:"occurredAt"`
	RelatedTrailID *uint64   `gorm:"index" json:"relatedTrailId,omitempty"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

func (KeyOperationTrace) TableName() string { return "enterprise_key_operation_traces" }

type CrossTeamAuthorizationSnapshot struct {
	ID             uint64    `gorm:"primaryKey" json:"id"`
	WorkspaceID    uint64    `gorm:"index" json:"workspaceId"`
	ProjectID      *uint64   `gorm:"index" json:"projectId,omitempty"`
	SnapshotAt     time.Time `gorm:"index" json:"snapshotAt"`
	SourceTeam     string    `gorm:"size:128;not null" json:"sourceTeam"`
	TargetTeam     string    `gorm:"size:128;not null" json:"targetTeam"`
	GrantType      string    `gorm:"size:64;not null" json:"grantType"`
	ScopeSummary   string    `gorm:"type:text" json:"scopeSummary"`
	Temporality    string    `gorm:"size:64" json:"temporality"`
	DelegationFlag bool      `gorm:"not null;default:false" json:"delegationFlag"`
	RiskHint       string    `gorm:"type:text" json:"riskHint"`
	TrendLabel     string    `gorm:"size:128" json:"trendLabel"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

func (CrossTeamAuthorizationSnapshot) TableName() string {
	return "enterprise_cross_team_authorization_snapshots"
}

type GovernanceRiskEvent struct {
	ID                uint64               `gorm:"primaryKey" json:"id"`
	WorkspaceID       uint64               `gorm:"index" json:"workspaceId"`
	ProjectID         *uint64              `gorm:"index" json:"projectId,omitempty"`
	RiskType          string               `gorm:"size:64;not null" json:"riskType"`
	Severity          string               `gorm:"size:32;not null" json:"severity"`
	SubjectSummary    string               `gorm:"type:text" json:"subjectSummary"`
	ScopeSummary      string               `gorm:"type:text" json:"scopeSummary"`
	TriggerReason     string               `gorm:"type:text" json:"triggerReason"`
	Status            GovernanceRiskStatus `gorm:"size:32;not null;index" json:"status"`
	RecommendedAction string               `gorm:"type:text" json:"recommendedAction"`
	Owner             string               `gorm:"size:128" json:"owner"`
	FirstSeenAt       time.Time            `gorm:"index" json:"firstSeenAt"`
	LastSeenAt        time.Time            `gorm:"index" json:"lastSeenAt"`
	CreatedAt         time.Time            `json:"createdAt"`
	UpdatedAt         time.Time            `json:"updatedAt"`
}

func (GovernanceRiskEvent) TableName() string { return "enterprise_governance_risk_events" }

type GovernanceCoverageSnapshot struct {
	ID                   uint64    `gorm:"primaryKey" json:"id"`
	WorkspaceID          uint64    `gorm:"index" json:"workspaceId"`
	ProjectID            *uint64   `gorm:"index" json:"projectId,omitempty"`
	SnapshotAt           time.Time `gorm:"index" json:"snapshotAt"`
	CoverageDomain       string    `gorm:"size:64;not null" json:"coverageDomain"`
	CoverageRate         float64   `gorm:"not null;default:0" json:"coverageRate"`
	StatusBreakdown      string    `gorm:"type:text" json:"statusBreakdown"`
	MissingReasonSummary string    `gorm:"type:text" json:"missingReasonSummary"`
	ConfidenceLevel      string    `gorm:"size:32;not null" json:"confidenceLevel"`
	TrendSummary         string    `gorm:"type:text" json:"trendSummary"`
	Owner                string    `gorm:"size:128" json:"owner"`
	CreatedAt            time.Time `json:"createdAt"`
	UpdatedAt            time.Time `json:"updatedAt"`
}

func (GovernanceCoverageSnapshot) TableName() string {
	return "enterprise_governance_coverage_snapshots"
}

type GovernanceReportPackage struct {
	ID                uint64                 `gorm:"primaryKey" json:"id"`
	WorkspaceID       uint64                 `gorm:"index" json:"workspaceId"`
	ProjectID         *uint64                `gorm:"index" json:"projectId,omitempty"`
	ReportType        string                 `gorm:"size:64;not null" json:"reportType"`
	Title             string                 `gorm:"size:128;not null" json:"title"`
	AudienceType      string                 `gorm:"size:64;not null" json:"audienceType"`
	TimeRange         string                 `gorm:"size:128" json:"timeRange"`
	SummarySection    string                 `gorm:"type:text" json:"summarySection"`
	DetailSection     string                 `gorm:"type:text" json:"detailSection"`
	AttachmentCatalog string                 `gorm:"type:text" json:"attachmentCatalog"`
	VisibilityPolicy  string                 `gorm:"type:text" json:"visibilityPolicy"`
	GeneratedAt       time.Time              `gorm:"index" json:"generatedAt"`
	GeneratedBy       uint64                 `gorm:"index;not null" json:"generatedBy"`
	Status            GovernanceReportStatus `gorm:"size:32;not null;index" json:"status"`
	CreatedAt         time.Time              `json:"createdAt"`
	UpdatedAt         time.Time              `json:"updatedAt"`
}

func (GovernanceReportPackage) TableName() string { return "enterprise_governance_report_packages" }

type ExportRecord struct {
	ID             uint64    `gorm:"primaryKey" json:"id"`
	PackageID      uint64    `gorm:"index;not null" json:"packageId"`
	ExportType     string    `gorm:"size:64;not null" json:"exportType"`
	AudienceScope  string    `gorm:"size:128;not null" json:"audienceScope"`
	ContentLevel   string    `gorm:"size:64;not null" json:"contentLevel"`
	Result         string    `gorm:"size:32;not null" json:"result"`
	AuditReference string    `gorm:"size:128" json:"auditReference"`
	ExportedAt     time.Time `gorm:"index" json:"exportedAt"`
	ExportedBy     uint64    `gorm:"index;not null" json:"exportedBy"`
	CreatedAt      time.Time `json:"createdAt"`
	UpdatedAt      time.Time `json:"updatedAt"`
}

func (ExportRecord) TableName() string { return "enterprise_export_records" }

type DeliveryArtifact struct {
	ID                uint64                 `gorm:"primaryKey" json:"id"`
	WorkspaceID       uint64                 `gorm:"index" json:"workspaceId"`
	ProjectID         *uint64                `gorm:"index" json:"projectId,omitempty"`
	ArtifactType      string                 `gorm:"size:64;not null" json:"artifactType"`
	Title             string                 `gorm:"size:128;not null" json:"title"`
	VersionScope      string                 `gorm:"size:128;not null" json:"versionScope"`
	EnvironmentScope  string                 `gorm:"size:128;not null" json:"environmentScope"`
	OwnerRole         string                 `gorm:"size:128;not null" json:"ownerRole"`
	ApplicabilityNote string                 `gorm:"type:text" json:"applicabilityNote"`
	Status            DeliveryArtifactStatus `gorm:"size:32;not null;index" json:"status"`
	UpdatedAt         time.Time              `json:"updatedAt"`
	CreatedAt         time.Time              `json:"createdAt"`
}

func (DeliveryArtifact) TableName() string { return "enterprise_delivery_artifacts" }

type DeliveryReadinessBundle struct {
	ID                  uint64                      `gorm:"primaryKey" json:"id"`
	WorkspaceID         uint64                      `gorm:"index" json:"workspaceId"`
	ProjectID           *uint64                     `gorm:"index" json:"projectId,omitempty"`
	Name                string                      `gorm:"size:128;not null" json:"name"`
	TargetEnvironment   string                      `gorm:"size:128;not null" json:"targetEnvironment"`
	TargetAudience      string                      `gorm:"size:128;not null" json:"targetAudience"`
	ArtifactSummary     string                      `gorm:"type:text" json:"artifactSummary"`
	ChecklistStatus     string                      `gorm:"size:64" json:"checklistStatus"`
	MissingItems        string                      `gorm:"type:text" json:"missingItems"`
	ReadinessConclusion DeliveryReadinessConclusion `gorm:"size:32;not null;index" json:"readinessConclusion"`
	UpdatedAt           time.Time                   `json:"updatedAt"`
	CreatedAt           time.Time                   `json:"createdAt"`
}

func (DeliveryReadinessBundle) TableName() string { return "enterprise_delivery_readiness_bundles" }

type DeliveryChecklistItem struct {
	ID                  uint64     `gorm:"primaryKey" json:"id"`
	BundleID            uint64     `gorm:"index;not null" json:"bundleId"`
	CheckItem           string     `gorm:"size:255;not null" json:"checkItem"`
	Category            string     `gorm:"size:64;not null" json:"category"`
	Owner               string     `gorm:"size:128;not null" json:"owner"`
	EvidenceRequirement string     `gorm:"type:text" json:"evidenceRequirement"`
	Status              string     `gorm:"size:32;not null" json:"status"`
	Remark              string     `gorm:"type:text" json:"remark"`
	CompletedAt         *time.Time `json:"completedAt,omitempty"`
	CreatedAt           time.Time  `json:"createdAt"`
	UpdatedAt           time.Time  `json:"updatedAt"`
}

func (DeliveryChecklistItem) TableName() string { return "enterprise_delivery_checklist_items" }

type GovernanceActionItem struct {
	ID                uint64                 `gorm:"primaryKey" json:"id"`
	WorkspaceID       uint64                 `gorm:"index" json:"workspaceId"`
	ProjectID         *uint64                `gorm:"index" json:"projectId,omitempty"`
	SourceType        string                 `gorm:"size:64;not null" json:"sourceType"`
	SourceRef         string                 `gorm:"size:128;not null" json:"sourceRef"`
	Title             string                 `gorm:"size:255;not null" json:"title"`
	Priority          string                 `gorm:"size:32;not null" json:"priority"`
	Owner             string                 `gorm:"size:128" json:"owner"`
	Status            GovernanceActionStatus `gorm:"size:32;not null;index" json:"status"`
	ResolutionSummary string                 `gorm:"type:text" json:"resolutionSummary"`
	DueAt             *time.Time             `json:"dueAt,omitempty"`
	CreatedAt         time.Time              `json:"createdAt"`
	UpdatedAt         time.Time              `json:"updatedAt"`
}

func (GovernanceActionItem) TableName() string { return "enterprise_governance_action_items" }

File: frontend/src/services/enterprisePolish.ts
import { fetchJSON } from '@/services/api/client';

type UnknownRecord = Record<string, unknown>;
type QueryValue = string | number | boolean | undefined | null;

const isRecord = (value: unknown): value is UnknownRecord =>
  typeof value === 'object' && value !== null;

const toText = (value: unknown): string | undefined => {
  if (typeof value === 'string') {
    const trimmed = value.trim();
    return trimmed.length > 0 ? trimmed : undefined;
  }
  if (typeof value === 'number' && Number.isFinite(value)) {
    return String(value);
  }
  return undefined;
};

const toNumber = (value: unknown): number | undefined => {
  if (typeof value === 'number' && Number.isFinite(value)) {
    return value;
  }
  if (typeof value === 'string') {
    const parsed = Number(value.trim());
    return Number.isFinite(parsed) ? parsed : undefined;
  }
  return undefined;
};

const toStringArray = (value: unknown): string[] => {
  if (!Array.isArray(value)) {
    return [];
  }
  return value.map((item) => toText(item)).filter((item): item is string => Boolean(item));
};

const toQueryString = (query: Record<string, QueryValue>) => {
  const params = new URLSearchParams();
  Object.entries(query).forEach(([key, value]) => {
    if (value === undefined || value === null || value === '') return;
    params.set(key, String(value));
  });
  return params.toString();
};

const withQuery = (path: string, query: Record<string, QueryValue>) => {
  const queryString = toQueryString(query);
  return queryString ? `${path}?${queryString}` : path;
};

const normalizeListResponse = <T>(value: unknown, mapper: (record: UnknownRecord) => T) => {
  if (Array.isArray(value)) {
    return { items: value.filter(isRecord).map(mapper) };
  }
  const record = isRecord(value) ? value : {};
  const items = Array.isArray(record.items) ? record.items : [];
  return { items: items.filter(isRecord).map(mapper) };
};

export type PermissionTrail = {
  id: string;
  subjectType?: string;
  subjectRef?: string;
  changeType?: string;
  beforeState?: string;
  afterState?: string;
  authorizationBasis?: string;
  evidenceCompleteness?: string;
  changedAt?: string;
};

export type KeyOperationTrace = {
  id: string;
  actorRef?: string;
  operationType?: string;
  targetType?: string;
  targetRef?: string;
  contextSummary?: string;
  riskLevel?: string;
  outcome?: string;
  occurredAt?: string;
};

export type GovernanceCoverageSnapshot = {
  id: string;
  coverageDomain?: string;
  coverageRate?: number;
  statusBreakdown?: string[];
  missingReasonSummary?: string;
  confidenceLevel?: string;
  trendSummary?: string;
  snapshotAt?: string;
};

export type GovernanceActionItem = {
  id: string;
  sourceType?: string;
  sourceRef?: string;
  title: string;
  priority?: string;
  owner?: string;
  status?: string;
  resolutionSummary?: string;
};

export type GovernanceReportPackage = {
  id: string;
  reportType?: string;
  title: string;
  audienceType?: string;
  timeRange?: string;
  summarySection?: string;
  detailSection?: string;
  attachmentCatalog: string[];
  status?: string;
};

export type ExportRecord = {
  id: string;
  exportType?: string;
  audienceScope?: string;
  contentLevel?: string;
  result?: string;
  exportedAt?: string;
};

export type DeliveryArtifact = {
  id: string;
  artifactType?: string;
  title: string;
  versionScope?: string;
  environmentScope?: string;
  ownerRole?: string;
  status?: string;
  applicabilityNote?: string;
};

export type DeliveryReadinessBundle = {
  id: string;
  name: string;
  targetEnvironment?: string;
  targetAudience?: string;
  artifactSummary?: string;
  checklistStatus?: string;
  missingItems: string[];
  readinessConclusion?: string;
};

export type DeliveryChecklistItem = {
  id: string;
  checkItem: string;
  category?: string;
  owner?: string;
  evidenceRequirement?: string;
  status?: string;
  remark?: string;
  completedAt?: string;
};

export type EnterpriseAuditEvent = {
  id: string;
  action?: string;
  targetType?: string;
  targetRef?: string;
  outcome?: string;
  occurredAt?: string;
};

export type CreateGovernanceReportPayload = {
  workspaceId: number;
  reportType: string;
  title: string;
  audienceType: string;
  timeRange?: string;
  visibilityPolicy?: string;
};

export type CreateExportRecordPayload = {
  audienceScope: string;
  contentLevel: string;
  exportType: string;
};

const mapPermissionTrail = (record: UnknownRecord): PermissionTrail => ({
  id: toText(record.id) || 'unknown-trail',
  subjectType: toText(record.subjectType),
  subjectRef: toText(record.subjectRef),
  changeType: toText(record.changeType),
  beforeState: toText(record.beforeState),
  afterState: toText(record.afterState),
  authorizationBasis: toText(record.authorizationBasis),
  evidenceCompleteness: toText(record.evidenceCompleteness),
  changedAt: toText(record.changedAt)
});

const mapKeyOperation = (record: UnknownRecord): KeyOperationTrace => ({
  id: toText(record.id) || 'unknown-operation',
  actorRef: toText(record.actorRef),
  operationType: toText(record.operationType),
  targetType: toText(record.targetType),
  targetRef: toText(record.targetRef),
  contextSummary: toText(record.contextSummary),
  riskLevel: toText(record.riskLevel),
  outcome: toText(record.outcome),
  occurredAt: toText(record.occurredAt)
});

const mapCoverage = (record: UnknownRecord): GovernanceCoverageSnapshot => ({
  id: toText(record.id) || 'unknown-coverage',
  coverageDomain: toText(record.coverageDomain),
  coverageRate: toNumber(record.coverageRate),
  statusBreakdown: toStringArray(record.statusBreakdown),
  missingReasonSummary: toText(record.missingReasonSummary),
  confidenceLevel: toText(record.confidenceLevel),
  trendSummary: toText(record.trendSummary),
  snapshotAt: toText(record.snapshotAt)
});

const mapActionItem = (record: UnknownRecord): GovernanceActionItem => ({
  id: toText(record.id) || 'unknown-action',
  sourceType: toText(record.sourceType),
  sourceRef: toText(record.sourceRef),
  title: toText(record.title) || '未命名待办',
  priority: toText(record.priority),
  owner: toText(record.owner),
  status: toText(record.status),
  resolutionSummary: toText(record.resolutionSummary)
});

const mapReport = (record: UnknownRecord): GovernanceReportPackage => ({
  id: toText(record.id) || 'unknown-report',
  reportType: toText(record.reportType),
  title: toText(record.title) || '未命名报表',
  audienceType: toText(record.audienceType),
  timeRange: toText(record.timeRange),
  summarySection: toText(record.summarySection),
  detailSection: toText(record.detailSection),
  attachmentCatalog: toStringArray(record.attachmentCatalog),
  status: toText(record.status)
});

const mapExportRecord = (record: UnknownRecord): ExportRecord => ({
  id: toText(record.id) || 'unknown-export',
  exportType: toText(record.exportType),
  audienceScope: toText(record.audienceScope),
  contentLevel: toText(record.contentLevel),
  result: toText(record.result),
  exportedAt: toText(record.exportedAt)
});

const mapDeliveryArtifact = (record: UnknownRecord): DeliveryArtifact => ({
  id: toText(record.id) || 'unknown-artifact',
  artifactType: toText(record.artifactType),
  title: toText(record.title) || '未命名材料',
  versionScope: toText(record.versionScope),
  environmentScope: toText(record.environmentScope),
  ownerRole: toText(record.ownerRole),
  status: toText(record.status),
  applicabilityNote: toText(record.applicabilityNote)
});

const mapBundle = (record: UnknownRecord): DeliveryReadinessBundle => ({
  id: toText(record.id) || 'unknown-bundle',
  name: toText(record.name) || '未命名交付包',
  targetEnvironment: toText(record.targetEnvironment),
  targetAudience: toText(record.targetAudience),
  artifactSummary: toText(record.artifactSummary),
  checklistStatus: toText(record.checklistStatus),
  missingItems: toStringArray(record.missingItems),
  readinessConclusion: toText(record.readinessConclusion)
});

const mapChecklist = (record: UnknownRecord): DeliveryChecklistItem => ({
  id: toText(record.id) || 'unknown-check',
  checkItem: toText(record.checkItem) || '未命名检查项',
  category: toText(record.category),
  owner: toText(record.owner),
  evidenceRequirement: toText(record.evidenceRequirement),
  status: toText(record.status),
  remark: toText(record.remark),
  completedAt: toText(record.completedAt)
});

const mapAuditEvent = (record: UnknownRecord): EnterpriseAuditEvent => ({
  id: toText(record.id) || 'unknown-audit',
  action: toText(record.action),
  targetType: toText(record.targetType),
  targetRef: toText(record.targetRef),
  outcome: toText(record.outcome),
  occurredAt: toText(record.occurredAt)
});

export const listPermissionTrails = async () =>
  normalizeListResponse(
    await fetchJSON<unknown>('/enterprise/audit/permission-trails'),
    mapPermissionTrail
  );

export const listKeyOperations = async () =>
  normalizeListResponse(await fetchJSON<unknown>('/enterprise/audit/key-operations'), mapKeyOperation);

export const listGovernanceCoverage = async () =>
  normalizeListResponse(await fetchJSON<unknown>('/enterprise/governance/coverage'), mapCoverage);

export const listGovernanceActionItems = async () =>
  normalizeListResponse(await fetchJSON<unknown>('/enterprise/governance/action-items'), mapActionItem);

export const listGovernanceReports = async () =>
  normalizeListResponse(await fetchJSON<unknown>('/enterprise/reports'), mapReport);

export const createGovernanceReport = async (payload: CreateGovernanceReportPayload) =>
  mapReport(
    (await fetchJSON<unknown>('/enterprise/reports', {
      method: 'POST',
      body: JSON.stringify(payload)
    })) as UnknownRecord
  );

export const createExportRecord = async (reportId: string, payload: CreateExportRecordPayload) =>
  mapExportRecord(
    (await fetchJSON<unknown>(`/enterprise/reports/${reportId}/exports`, {
      method: 'POST',
      body: JSON.stringify(payload)
    })) as UnknownRecord
  );

export const listDeliveryArtifacts = async () =>
  normalizeListResponse(await fetchJSON<unknown>('/enterprise/delivery/artifacts'), mapDeliveryArtifact);

export const listDeliveryBundles = async () =>
  normalizeListResponse(await fetchJSON<unknown>('/enterprise/delivery/bundles'), mapBundle);

export const listDeliveryChecklist = async (bundleId: string) =>
  normalizeListResponse(
    await fetchJSON<unknown>(withQuery(`/enterprise/delivery/bundles/${bundleId}/checklists`, {})),
    mapChecklist
  );

export const listEnterpriseAuditEvents = async (query: Record<string, QueryValue>) =>
  normalizeListResponse(
    await fetchJSON<unknown>(withQuery('/audit/enterprise/events', query)),
    mapAuditEvent
  );
